packages/@lwc/engine/src/framework/vm.ts

```diff
@@ -641,6 +641,19 @@ export function getShadowRootVM(root: ShadowRoot): VM {
     return getHiddenField(root, ViewModelReflection) as VM;
 }
 
+function hasDifferentIdentities(a: VNode | null, b: VNode | null): boolean {
+    if (a === b) {
+        // cached vnodes or null will qualify here
+        return false;
+    }
+    if (isNull(a) || isNull(b)) {
+        // replace or insertion
+        return true;
+    }
+    // keys must be different otherwise they represent the same element
```

Comment:
```suggestion
    // keys must be different otherwise they represent the same element
    // if keys are different, it means the vnodes are representing a different element
    // if the keys are the same, it means the vnodes are representing the same element
```